physical/raft/raft.go

```diff
@@ -1622,10 +1617,28 @@ func (b *RaftBackend) applyLog(ctx context.Context, command *LogData) error {
 		resp = chunkedSuccess.Response
 	}
 
-	if resp, ok := resp.(*FSMApplyResponse); !ok || !resp.Success {
+	fsmar, ok := resp.(*FSMApplyResponse)
+	if !ok || !fsmar.Success {
 		return errors.New("could not apply data")
 	}
 
+	// populate command with our results
+	if fsmar.EntrySlice == nil {
+		return errors.New("entries on FSM response were empty")
+	}
+
+	for i, logOp := range command.Operations {
+		if logOp.OpType == getOp {
+			fsmEntry := fsmar.EntrySlice[i]
+
+			// this should always be true because the entries in the slice were created in the same order as
+			// the command operations.
+			if logOp.Key == fsmEntry.Key && len(fsmEntry.Value) > 0 {
```

Comment:
I suppose if you're going to worry about a nil EntrySlice, you should also maybe return an error when when the key doesn't match.  Up to you.